### DISCUSSION TITLE
Peers cannot reach each other within or between CIDRs

### DISCUSSION THREAD

**Discussion by bowen-song** [tonarino/innernet#116] [Q&A] [answered]
Hi!
Just want to start off by saying this is a very neat project, and thanks!

Now onto the problem I'm having.
Setup: 
`sudo inn list --tree`
```
bween
  10.42.0.0/16 bween
    10.42.0.1/32 innernet-server
    | ◉ 10.42.0.1: innernet-server 
    10.42.1.0/24 users
    | ◉ 10.42.1.1: archxps
    | ◯ 10.42.1.2: mbp 
    10.42.2.0/24 servers
    | ◯ 10.42.2.1: media-server

```
`sudo inn list-associations bween`
```
[*] Fetching CIDRs
[*] Fetching associations
1: users <=> servers
```

I've followed the documentation to the best of my understanding and there weren't any errors while installing, but when attempting to connect between peers, the only connection that works is between peer and "innernet server" located at 10.42.0.1; all other connections never make it.

As an example from `archxps`:
`ping innernet-server.bween.wg` -> okay
`ping mbp.bween.wg` -> no connection
`ping media-server.bween.wg` -> no connection

I've tried pinging with the direct ip address and that hasn't worked either so I don't believe it's a DNS problem.

For additional details, `innernet-server` is hosted on a vps, and has a public facing wg port, while all users/servers are behind a local OpenWrt router with no open/listen ports. 

I'm not sure if the NAT holepunching method is the same with innernet as it is with Tailscale, but in case it helps troubleshoot, I have run Tailscale on these machines with no problems so I think I must be setting something up wrong (I just can't figure out what it is!!).

Thanks!

Edit: Just noticed that the output of `sudo inn list --tree` shows two filled circles for `archxps` and `innernet-server` whereas the others are empty. When I run `sudo inn list --tree` on `mbp`, there are two filled circles for `mbp` and `innernet-server`. This coincides with the connections I'm able to make. Still puzzled by what the problem is though.

**Comment by mcginty** [comment, marked as answer]
Answered in issue #117 :).
